ecgtools/core.py

```diff
@@ -259,8 +261,15 @@ def build(
             'aggregations': aggregations,
         }
 
-        filelist = self.filelist or self._get_filelist_from_dirs()
-        df = parse_files_attributes(filelist, self.parser, self.lazy, self.nbatches)
+        if len(self.filelist) == 0:
+            filelist = self.filelist or self._get_filelist_from_dirs()
+        else:
+            filelist = self.filelist
+
+        if local_attrs is None:
+            local_attrs = {}
```

Comment:
Is there a reason why `{}` should just be the default value of `local_attrs` (instead of `None`)?  (Or, another way of asking that, is there a good reason to default `local_attrs` to `None`?)

Reply:
Thanks @kmpaul , I changed the default and removed these lines.